lib/CBonusTypeHandler.cpp

```diff
@@ -84,6 +84,13 @@ std::string CBonusTypeHandler::bonusToString(const std::shared_ptr<Bonus> & bonu
 	if (text.find("${subtype.spell}") != std::string::npos && bonus->subtype.as<SpellID>().hasValue())
 		boost::algorithm::replace_all(text, "${subtype.spell}", bonus->subtype.as<SpellID>().toSpell()->getNameTranslated());
 
+	if (text.find("${subtype.spellSchool}") != std::string::npos && bonus->subtype.as<SpellSchool>().hasValue())
+	{
+		auto school = bonus->subtype.as<SpellSchool>();
+		auto schoolName = VLC->generaltexth->zelp[school == SpellSchool::ANY ? 458 : 454 + school].first;
+		boost::algorithm::replace_all(text, "${subtype.spellSchool}", schoolName);
+	}
+
```

Comment:
This won't work in Ukrainian and probably rest of Slavic languages due to more complex grammatical cases:
Water Spells: Магія Води / Закляття Води
Immunity to Water spells: Іммунітет до магії Води / Іммунітет до заклять Води
"Magiya" became "Magiyi", "Zaklyattya" became "Zaklyat`"

And even in English this would result in somewhat awkward "Immunity to Water Spells" - why "Spells" are capitalized?

Proposition: add custom descriptions (not sure about names) for all bonuses that have spell school as subtype.

For this you would need:
- In `CBonusTypeHandler::loadItem`, for these bonuses instead of registering description string, you will need to register 4 additional string for all 4 spell schools
- In `CBonusTypeHandler::bonusToString`, if spell has spell school that is not "all" as subtype, use specific translation instead of generic one.

And please check also bonuses `SPELL_DAMAGE_REDUCTION` and `NEGATIVE_EFFECTS_IMMUNITY`. I think they also may need per-school translations

Reply:
Added `.specific`-Identifier.

Should also work with `SPELL_DAMAGE_REDUCTION` and `NEGATIVE_EFFECTS_IMMUNITY`. If translations will be added (not part of this PR).